### ISSUE TITLE
Deprecation warning from SequentialLR

### ISSUE THREAD

**Issue by xkszltl** [pytorch/pytorch#76113]
### 🐛 Describe the bug

This is probably from sub-scheduler when SequentialLR calls them with epoch arg.

```
/usr/local/lib/python3.6/dist-packages/torch/optim/lr_scheduler.py:154: UserWarning: The epoch parameter in `scheduler.step()` was not necessary and is being deprecated where possible. Please use `scheduler.step()` to step the scheduler. During the deprecation, if epoch is different from None, the closed form is used instead of the new chainable form, where available. Please open an issue if you are unable to replicate your use case: https://github.com/pytorch/pytorch/issues/new/choose.
  warnings.warn(EPOCH_DEPRECATION_WARNING, UserWarning)
```

Related code:
- https://github.com/pytorch/pytorch/blob/29a45f00097664d12921c8d13e387bee4b835665/torch/optim/lr_scheduler.py#L154
- https://github.com/pytorch/pytorch/blob/29a45f00097664d12921c8d13e387bee4b835665/torch/optim/lr_scheduler.py#L632

Comment should also be fixed:
- https://github.com/pytorch/pytorch/blob/29a45f00097664d12921c8d13e387bee4b835665/torch/optim/lr_scheduler.py#L1276

For compatibility with 3rd-party sub-scheduler who's still relying on epoch arg, you may:
- Add a member switch to temporarily suppress this warnings when called directly by SequentialLR, and revert it at the end of step().
- Or whitelist pytorch's own types with warnings in SequentialLR to use non-epoch signature.
You may also need to check the type of `self` before printing that warning, to keep it quiet on derived types.

### Versions

Commit 29a45f00097664d12921c8d13e387bee4b835665

cc @ezyang @gchanan @zou3519

**Comment by xkszltl** [comment]
Another related issue:
SequentialLR cannot be nested because its step() does not take that additional 0.

**Comment by senarvi** [comment]
Duplicate of [#130018](https://github.com/pytorch/pytorch/issues/130018)?

**Comment by xkszltl** [comment]
It certainly cannot duplicate something from 2+ years later, but any fixes are still welcome.

**Comment by zeshengzong** [comment]
Hi, this already fixed in #149392 and this issue can be closed, thanks!

**Comment by xkszltl** [comment]
@zeshengzong Does it close https://github.com/pytorch/pytorch/issues/130018 as well?

**Comment by zeshengzong** [comment]
Thanks! I left a comment in the issue.
